ajax/impact.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2019 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+const DELTA_ACTION_ADD    = 1;
+const DELTA_ACTION_UPDATE = 2;
+const DELTA_ACTION_DELETE = 3;
+
+$AJAX_INCLUDE = 1;
+include ('../inc/includes.php');
+
+// Send UTF8 Headers
+header("Content-Type: application/json; charset=UTF-8");
+Html::header_nocache();
+
+Session::checkLoginUser();
+
+switch ($_SERVER['REQUEST_METHOD']) {
+   // GET request: build the impact graph for a given asset
+   case 'GET':
```

Comment:
Different HTTP method will do things very different? I'm really not fan

Reply:
This seems fine to me, `GET` allow to access impact data and `POST` to send a new impact payload (update).

Without this we would need another parameter like "action" to differentiate our two possible actions (or split them in two ajax files).
Using the HTTP method seems a simple and clean way to do it.

@cedric-anne @orthagh, thoughts ?

Reply:
I usually do it the same way, but it is not how it is done in GLPI right now. So I do not really have an opinion about this.